src/sentry/api/endpoints/check_am2_compatibility.py

```diff
@@ -23,18 +24,26 @@ class CheckAM2CompatibilityEndpoint(Endpoint):
     def get(self, request: Request) -> Response:
         try:
             org_id = request.GET.get("orgId")
-            check_status = get_check_status(org_id)
-            if check_status == CheckStatus.DONE:
-                results = get_check_results(org_id)
-                # In case the state is done, but we didn't find a valid value in cache, we have a problem.
-                if results is None:
-                    raise Exception("the check status is done in cache but there are no results.")
-
-                return Response({"status": CheckStatus.DONE.value, **results}, status=200)
-            elif check_status == CheckStatus.IN_PROGRESS:
-                return Response({"status": CheckStatus.IN_PROGRESS.value}, status=202)
-            elif check_status == CheckStatus.ERROR:
-                raise Exception("the asynchronous task had an internal error.")
+            refresh = request.GET.get("refresh") == "true"
```

Comment:
should we change the frontend to send this param? Also be should probably use cache control header for this.

Reply:
We would need a button to trigger the refresh, but it is only shown when the results are there.

Regarding the header, yeah that would be the best but I was aiming for a quick solution since this is a temporary endpoint that will likely last for a definite amount of time.